<a href="https://colab.research.google.com/github/Carlosmagnobrusil/CarlosBrusil_projects/blob/main/Semana_2/Titanic.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Datos Titanic**
## *Creado por: Carlos Brusil*



*   Fecha de Creación: 14 de enero de 2022 
*   Fechad de actualización: 14 de enero de 2022
*   Descripción: Practicar los conceptos de los metodos pandas aprendidoshasta el final de la semana 2 

### Carga de datos 

In [1]:
from google.colab import drive
drive.mount('/content/drive') # uno my Drive al nuevo cuaderno


Mounted at /content/drive


In [2]:
import pandas as pd #importo la libreria panda 
titanic =  '/content/drive/MyDrive/Coding_Dojo/Semana 2/Pandas Data science/Data /titanic.csv' #Nombro a la base
titanic_df = pd.read_csv(titanic) #le doy un nombre a la trama 
#de datos  ene ste caso el sugerido en la tarea como no tiene titulos el archivo txt uso el metodo header y que mantega los 
#valores vacios con na 
titanic_df.head() #Observo que los datos hayan sido leidos de forma correcta
#titanic_df.info()


PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [3]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


### Porcentaje de supervivientes 

In [4]:
#obs_tot = 891 # desde una aproximacion  podria haber creado una variable del total de personas en la base
#sobrevivientes = titanic_df['Survived'].sum() #como un segundo paso a este enfoque contaria el numero de sobrevivientes
#en este caso como las personas que han sobrevivio tomn el valor de 1 realizo un metodo de suma para la solumna de
#sobrevivientes 
#porcentaje= sobrevivientes/obs_tot #realizo la division que mentregara el porcentaje de perdsonas sobrevivientes
#porcentaje muestro el porcentaje obtenido 
sobrevivientes = titanic_df['Survived'].mean() #Una manera mas corta fue usar el promedio, cabe aclarar que 
#en este caso fue posible debido a que los datos solo eran 1 y 0 caso contarrio deberia haber cambiado el enfoque  
print ('El',str('{0:.2f}'.format((sobrevivientes)*100)),'%', 'de las personas embarcadas a sobrevivido')
#en la linea de codigo superior lo que hago es crear una cadena que muestre el porcentaje con unicamente dos decimales 
# para mostrar el resulatado 


El 38.38 % de las personas embarcadas a sobrevivido


### Numero de Sobrevivientes por sexo 

In [5]:
titanic_df.groupby(['Sex'])['Survived'].sum() #Uso la funcion de agrupamiento con el parametro del sexo y el metodo de 
#suma para que realize un conteo de las personas que sobrevivieron segun su sexo 

Sex
female    233
male      109
Name: Survived, dtype: int64

#### Sobrevivieron **233** personas de sexo **femenino** y **109** personas de sexo **masculino** 

### Porcentaje de sobrevivientes por valor del pasaje 

In [6]:
obs_tot = 891 #fijo la variable del total de pasajeros u observaciones del data frame 
pasaje = titanic_df['Fare']<10 #fijo el filtro para pasajero de ticket de menos de 10 
sobre_pasaje = titanic_df[pasaje]['Survived'].sum()#sumo todos los pasajeros que sobrevivieron con un ticket de menos de 10
sobrevivientes_fare = sobre_pasaje/obs_tot
print('El',str('{0:.2f}'.format(sobrevivientes_fare*100)),'%','de personas que pagaron un boleto de menos 10 sobrevivieron' )

El 7.52 % de personas que pagaron un boleto de menos 10 sobrevivieron


### Edad promedio de quienes no sobrevivieron

In [7]:
no_sobrevivientes = titanic_df['Survived']==0 # defino un filtro que me diga quienes sobrevivieron o no
edad_nosobrev = titanic_df[no_sobrevivientes]['Age'].mean() #obtengo la media del filtro creado 
print('La edad promedio de los no sobrevivientes es de ', str('{0:.1f}'.format(edad_nosobrev)),'años')#imprimo la respuesta 

La edad promedio de los no sobrevivientes es de  30.6 años


###  Edad promedio de quienes sobrevivieron

In [8]:
sobrevivient = titanic_df['Survived']==1
edad_sobrev = titanic_df[sobrevivient]['Age'].mean()
print('La edad promedio de los sobrevivientes es de ', str('{0:.1f}'.format(edad_sobrev)),'años')

La edad promedio de los sobrevivientes es de  28.3 años


### edad promedio de quienes sobrevivieron y no sobrevivieron agrupados por sexo

In [9]:
round(titanic_df.groupby(['Survived','Sex'])['Age'].mean(),2)

Survived  Sex   
0         female    25.05
          male      31.62
1         female    28.85
          male      27.28
Name: Age, dtype: float64